content/blog/2018-12-25-darktable-2.6/2018-12-25-darktable-2.6.md

```diff
@@ -0,0 +1,335 @@
+from GIMP's "heal" tool), which adapts the source to the context
+where it is cloned. You don't need to clone *exactly* the right
+piece of image. Let's take a common example: a small defect in a
+not-completely-uniform sky:
+
+  <div style="text-align: center">
+![Retouch example]({filename}rt-original.png)
+  </div>
+
+A bad attempt to fix this with the clone tool would give:
+
+  <div style="text-align: center">
+![Retouch clone (with controls)]({filename}rt-clone.png)
+  </div>
+
+The piece of sky copied to mask the defect is a bit darker than the
+place where it's copied. It's not obvious while the controls are
+displayed on the image, but the final image is really bad:
+
+  <div style="text-align: center">
+![Retouch clone (without controls)]({filename}rt-clone-nocontrol.png)
+  </div>
+
+The same retouch with the new "heal" tool
+(![heal tool]({filename}rt-heal-icon.png)) gives this:
+
+  <div style="text-align: center">
+![Retouch heal (with controls)]({filename}rt-heal.png)
+  </div>
+
+This time, the final image is undistinguishable from a real sky
+without defect:
+
+  <div style="text-align: center">
+![Retouch heal (without controls)]({filename}rt-heal-nocontrol.png)
+  </div>
+
+Even when cloning pieces of images of totally different colors, the
+"heal" tool behaves surprisingly well. Let's push the module a bit:
+
+  <div style="text-align: center">
+![Retouch heal on different colors]({filename}rt-heal-red-white-blue.png)
+  </div>
+
+The white patch is cloned to the blue shirt, the blue shirt to the red
+one, and the red one to the white one. Each time, the local contrast
+is kept, but the overall color and luma of the patch is adapted to fit
+the destination.
+
+The basic "clone" tool (![clone tool]({filename}rt-clone-icon.png)) is
+still available for the rare cases where you would want it.
+
+#### Fill and blur, when you don't have anything to clone
+
+In addition to the "clone" and "heal" tools (which work only when
+you have a piece of image to duplicate on the spot to remove),
+retouch provides "fill" (![fill tool]({filename}rt-fill-icon.png))
+(pick a color, and fill a shape with this
+color) and "blur" (![blur tool]({filename}rt-blur-icon.png))
+ (apply a blur to soften a part of the image)
+tools.
+
+#### One activation, multiple strokes: continuous add
+
+Tools can be activated once for multiple strokes. Just use
+Control-click on one of the point, line, ellipse or path tool
+(instead of a normal click), and the tool will remain active until
+you explicitly deactivate it. This is very convenient when fixing
+many spots on the same image, compared to the previous flow where
+one had to re-click the tool's button for each stroke.
+
+#### Source patch visualization
+
+For "clone" and "heal" tools, each stroke consists in a source and a
+destination. With a simple click, one sets the destination and by
+default darktable guesses the location of the source. One can already
```

Comment:
Actually dt don't guess but has the source location hard-coded at first. For the circle & ellipse it depends on the radius, for the path & brush is a hard-coded number. Not sure if this is relevant to the end user, just commenting on it because I think that LR guess the "best" source when healing.

Reply:
Changed to "picks an arbitrary location for". Indeed "guess" may suggest a bit too much (I'd love a real auto-guess source feature actually, but that's another topic).